packages/auth/src/tenancy/context.js

```diff
@@ -53,6 +53,11 @@ exports.setTenantId = (
   // processed later in the chain
   tenantId = user.tenantId || header || tenantId
 
+  // Set the tenantId from the subdomain
+  if (!tenantId) {
+    tenantId = ctx.subdomains && ctx.subdomains[0]
```

Comment:
do you think we need a check here for when there is a user and tenant id but it doesn't match the tenant being requested? e.g. I'm logged in to `abc.budibase.app` but I navigate on over to `xyz.budibase.app`. I think atm it will silently ignore the `xyz` part and serve me data from `abc`. We'd probably want to redirect them away or fail the request

Reply:
Will it even reach this branch if that's the case? It should be set earlier on in this middleware, I think

Reply:
I think it's still an issue, here's an example where my session is scoped to `rpowell`
![Screenshot 2021-09-29 at 11 09 54](https://user-images.githubusercontent.com/8755148/135249135-4ecb893f-3362-4c16-a828-5f708cc98924.png)

But I nav to `disney` and budibase doesn't complain and still serves me `rpowell` data
![Screenshot 2021-09-29 at 11 10 06](https://user-images.githubusercontent.com/8755148/135249252-6364f03c-15a7-494a-8103-e9c9e9a2e995.png)

If the tenant doesn't match we should probably kick them back to account portal (which should send them to the correct tenant if they don't have account portal access)

Reply:
probably easiest to prevent this in the svelte app, and just do a sanity in the backend by throwing a 403